instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -72,22 +72,34 @@ def __init__(self, tracer, engine):
         listen(engine, "after_cursor_execute", self._after_cur_exec)
         listen(engine, "handle_error", self._handle_error)
 
+    def _operation_name(self, db_name, statement):
+        parts = []
+        if isinstance(statement, str):
+            parts.append(statement.split(" ")[0].split("\t")[0])
+        if db_name:
+            parts.append(db_name)
+        if not parts:
+            return self.vendor
+        return " ".join(parts)
+
     # pylint: disable=unused-argument
     def _before_cur_exec(self, conn, cursor, statement, *args):
+        attrs = {}
+        attrs, found = _get_attributes_from_url(conn.engine.url, attrs)
```

Comment:
```suggestion
        attrs, found = _get_attributes_from_url(conn.engine.url)
```
Single use functiong here, BTW.